libraries/AC_PID/AC_PID.cpp

```diff
@@ -215,13 +208,22 @@ void AC_PID::operator() (float p, float i, float d, float imaxval, float input_f
 }
 
 // calc_filt_alpha - recalculate the input filter alpha
-float AC_PID::get_filt_alpha() const
-{
-    if (is_zero(_filt_hz)) {
-        return 1.0f;
+float AC_PID::get_filt_alpha()
+{
+    if (!is_equal(_dt, _last_filt_alpha_dt) ||
+        !is_equal(_filt_hz.get(), _last_filt_alpha_filt_hz)) {
+        // current cached alpha value is out-of-date
+        if (is_zero(_filt_hz)) {
+            // clamp the alpha value
+            _filt_alpha = 1.0f;
+        } else {
+            // calculate alpha
+            const float rc = 1/(M_2PI*_filt_hz);
```

Comment:
`const float rc = (1/M_2PI)/*_filt_hz;` should be better as the compiler can then do the 1/M_2PI calculation for us, and while not a numerically stable transformation I think it's fine for the ranges we use it in.